core/LSU.cpp

```diff
@@ -452,12 +682,192 @@ namespace olympia
         // instruction issue arbitration should always succeed, even when flush happens.
         // Otherwise, assertion error is fired inside arbitrateInstIssue_()
     }
+    void LSU::replayReady_(const InstPtr & replay_inst_ptr)
+    {
+        ILOG("Replay inst ready " << replay_inst_ptr);
+        // We check in the ldst_queue as the instruction may not be in the replay queue
+        for (auto & load_store_info_ptr : ldst_inst_queue_)
+        {
+            auto & inst_ptr = load_store_info_ptr->getInstPtr();
+            if (inst_ptr == replay_inst_ptr)
+            {
+                if(load_store_info_ptr->getState() == LoadStoreInstInfo::IssueState::NOT_READY)
+                {
+                    load_store_info_ptr->setState(LoadStoreInstInfo::IssueState::READY);
+                }
+                auto issue_priority = load_store_info_ptr->getMemoryAccessInfoPtr()->getPhyAddrStatus()
+                                          ? LoadStoreInstInfo::IssuePriority::CACHE_PENDING
+                                          : LoadStoreInstInfo::IssuePriority::MMU_PENDING;
+                load_store_info_ptr->setPriority(issue_priority);
+            }
+        }
+        if (isReadyToIssueInsts_())
+        {
+            ILOG("replay ready issue");
+            uev_issue_inst_.schedule(sparta::Clock::Cycle(0));
+        }
+    }
+
+    void LSU::updateInstReplayReady_(const InstPtr &inst_ptr){
+        for(auto iter = replay_buffer_.begin(); iter != replay_buffer_.end(); iter++)
```

Comment:
Why is the `InstPtr` being passed here?  Shouldn't it be a `LoadStoreInfo` type?  If you change this (strong suggestion you do), you can stash the replay iterator in the `LoadStoreInfo` and perform one operation:
```
replay_buffer_->erase(load_store_info_->getReplayIterator());
```